### PR TITLE
chore: open private chat with global message

### PR DESCRIPTION
**Pull request by pravusjif**

### CODE REVIEW

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/ChatWidgetHUD/ChatHUDController.cs [hunk 1]
```diff
@@ -9,19 +9,21 @@ public class ChatHUDController : IDisposable
...
-        else
-        {
-            this.view = view;
-        }
+        this.view = view ?? ChatHUDView.Create();
 
         this.view.Initialize(this, onSendMessage);
+
+        this.view.OnPressPrivateMessage -= View_OnPressPrivateMessage;
+        this.view.OnPressPrivateMessage += View_OnPressPrivateMessage;
```
Comment (sandrade-dcl) [inline unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/ChatWidgetHUD/ChatHUDController.cs:21]: I think it is always a good practice to unsubscribe all the subscriptions. In this case we could do it in the Dispose method.

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/HUDController.cs [hunk 2]
```diff
@@ -143,28 +143,33 @@ public void ConfigureHUDElement(HUDElementID hudElementId, HUDConfiguration conf
                 break;
             case HUDElementID.WORLD_CHAT_WINDOW:
                 CreateHudElement<WorldChatWindowHUDController>(configuration, hudElementId);
-                worldChatWindowHud?.Initialize(ChatController.i, DCL.InitialSceneReferences.i?.mouseCatcher);
-                taskbarHud?.AddWorldChatWindow(worldChatWindowHud);
+                if (worldChatWindowHud != null)
+                {
+                    worldChatWindowHud.Initialize(ChatController.i, DCL.InitialSceneReferences.i?.mouseCatcher);
+                    worldChatWindowHud.OnPressPrivateMessage -= OpenPrivateChatWindow;
+                    worldChatWindowHud.OnPressPrivateMessage += OpenPrivateChatWindow;
```
Comment (sandrade-dcl) [inline unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/HUDController.cs:150]: The same as before. In this case we could use the OnDestroy event to unsubscribtions.

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/HUDController.cs [hunk 3]
```diff
@@ -143,28 +143,33 @@ public void ConfigureHUDElement(HUDElementID hudElementId, HUDConfiguration conf
...
             case HUDElementID.FRIENDS:
                 CreateHudElement<FriendsHUDController>(configuration, hudElementId);
 
                 if (friendsHud != null)
                 {
                     friendsHud.Initialize(FriendsController.i, UserProfile.GetOwnUserProfile());
-                    friendsHud.OnPressWhisper -= FriendsHud_OnPressWhisper;
-                    friendsHud.OnPressWhisper += FriendsHud_OnPressWhisper;
+                    friendsHud.OnPressWhisper -= OpenPrivateChatWindow;
+                    friendsHud.OnPressWhisper += OpenPrivateChatWindow;
```
Comment (sandrade-dcl) [inline unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/HUDController.cs:172]: More about unsubscriptions.

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/PrivateChatWindow/PrivateChatWindowHUDController.cs [hunk 4]
```diff
@@ -22,6 +22,7 @@ public class PrivateChatWindowHUDController : IHUD
     public void Initialize(IChatController chatController)
     {
         view = PrivateChatWindowHUDView.Create(this);
+        view.OnPressBack -= View_OnPressBack;
         view.OnPressBack += View_OnPressBack;
```
Comment (sandrade-dcl) [inline unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/PrivateChatWindow/PrivateChatWindowHUDController.cs:26]: More about unsubscriptions.

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/WorldChatWindowHUD/WorldChatWindowHUDController.cs [hunk 5]
```diff
@@ -18,13 +18,17 @@ public class WorldChatWindowHUDController : IHUD
...
+
     public void Initialize(IChatController chatController, IMouseCatcher mouseCatcher)
     {
         view = WorldChatWindowHUDView.Create();
         view.controller = this;
 
         chatHudController = new ChatHUDController();
         chatHudController.Initialize(view.chatHudView, SendChatMessage);
+        chatHudController.OnPressPrivateMessage -= ChatHUDController_OnPressPrivateMessage;
+        chatHudController.OnPressPrivateMessage += ChatHUDController_OnPressPrivateMessage;
```
Comment (sandrade-dcl) [inline unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/WorldChatWindowHUD/WorldChatWindowHUDController.cs:31]: More about unsubscriptions.

**Review by sandrade-dcl — CHANGES_REQUESTED** [review]

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/WorldChatWindowHUD/WorldChatWindowHUDController.cs (hunk 5 above)
Comment (pravusjif) [reply unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/WorldChatWindowHUD/WorldChatWindowHUDController.cs:31]: Done

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/PrivateChatWindow/PrivateChatWindowHUDController.cs (hunk 4 above)
Comment (pravusjif) [reply unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/PrivateChatWindow/PrivateChatWindowHUDController.cs:26]: Done

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/HUDController.cs (hunk 3 above)
Comment (pravusjif) [reply unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/HUDController.cs:172]: Done

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/HUDController.cs (hunk 2 above)
Comment (pravusjif) [reply unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/HUDController.cs:150]: Done

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/ChatWidgetHUD/ChatHUDController.cs (hunk 1 above)
Comment (pravusjif) [reply unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/ChatWidgetHUD/ChatHUDController.cs:21]: Done

**Review by sandrade-dcl — APPROVED** [review]

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/ChatWidgetHUD/ChatEntry.cs [hunk 6]
```diff
@@ -89,26 +95,33 @@ public void Populate(Model chatEntryModel)
...
+        OnPress?.Invoke(model.otherUserId);
+    }
 
     public void SetFadeout(bool enabled)
     {
         if (!enabled)
         {
             group.alpha = 1;
-            this.enabled = false;
+            processFading = false;
```
Comment (BrianAmadori) [inline unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/ChatWidgetHUD/ChatEntry.cs]: maybe naming this as `fadeEnabled` is more clear? 

```suggestion
            fadeEnabled = false;
```

**Review by BrianAmadori — APPROVED** [review]

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/ChatWidgetHUD/ChatEntry.cs (hunk 6 above)
Comment (pravusjif) [reply unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/ChatWidgetHUD/ChatEntry.cs]: Done

**Pull request merged**